src/validator.rs

```diff
@@ -17,17 +15,28 @@ pub struct Validator {
     // uint256 minThreshold,
 }
 
-impl tokens::Detokenize for Validator {
-    /// Returns a `Validator` if the token's types match the fields.
-    fn from_tokens(tokens: Vec<Token>) -> Result<Validator, Error> {
-        match (tokens.get(0), tokens.get(1)) {
-            (Some(&Token::FixedBytes(ref first)), Some(&Token::FixedBytes(ref last))) => {
-                Ok(Validator {
-                    first_name: util::bytes_to_string(first),
-                    last_name: util::bytes_to_string(last),
-                })
-            }
-            _ => Err(ErrorKind::InvalidOutputType("Validator".to_string()).into()),
+type ValidatorTuple = (
+    ethabi::Hash,
+    ethabi::Hash,
+    ethabi::Hash,
+    String,
+    ethabi::Hash,
+    ethabi::Hash,
+    ethabi::Uint,
+    ethabi::Uint,
+    ethabi::Uint,
+    ethabi::Uint,
+);
+
+impl From<ValidatorTuple> for Validator {
+    fn from((first_name_h, last_name_h, ..): ValidatorTuple) -> Validator {
```

Comment:
Wow, I had no idea that Rust lets you unpack a tuple into function arguments like this.